specification/machinelearningservices/data-plane/Microsoft.MachineLearningServices/preview/2019-05-01/runHistory.json

```diff
@@ -0,0 +1,3042 @@
+        }
+      }
+    },
+    "/history/v1.0/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.MachineLearningServices/workspaces/{workspaceName}/experiments/{experimentName}": {
+      "get": {
+        "tags": [
+          "Experiments"
+        ],
+        "summary": "Get details of an Experiment.",
+        "description": "Get details of an Experiment with specific Experiment name.",
+        "operationId": "Experiments_Get",
+        "consumes": [],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/workspaceNameParameter"
+          },
+          {
+            "name": "experimentName",
+            "in": "path",
+            "description": "The experiment name.",
+            "required": true,
+            "type": "string"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The details of the Experiment are returned successfully.",
+            "schema": {
+              "$ref": "#/definitions/Experiment"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      },
+      "post": {
+        "tags": [
+          "Experiments"
+        ],
+        "summary": "Create an Experiment.",
+        "description": "Create a new Experiment.",
+        "operationId": "Experiments_Create",
+        "consumes": [],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/workspaceNameParameter"
+          },
+          {
+            "name": "experimentName",
+            "in": "path",
+            "description": "The experiment name.",
+            "required": true,
+            "type": "string"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The Experiment is created successfully.",
+            "schema": {
+              "$ref": "#/definitions/Experiment"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/history/v1.0/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.MachineLearningServices/workspaces/{workspaceName}/experiments": {
+      "get": {
```

Comment:
I think we discussed this in a previous meeting but can someone remind me why we landed on keeping both the Get and Query endpoints for Runs/Experiments/Metrics? I know there was some discussion about the Query as Post implementation being weird even though it gets us around the 413 failure mode but keeping both seems like the worst option.

Reply:
@thwagne At that point, we couldn't come to a conclusion on which API would be the best to go with therefore we kept both

1. HTTP get for fetch metrics api with filters passed in as query param would be ideal but we run into 413 failure in following scenarios
          a. because of  filter query params contributing to large URI length(seen in UX scenarios but most likely not in user cases)
          b. because of long continuation tokens returned by cosmosdb contributing to large URI length (sometimes).

2. HTTP post call with filters passed as payload to addresses the above issue but not ideal as per API review board suggestion

@johanste @andscho-msft any comments on how can we address this?

Reply:
@grmadala, I assume that you mean [414 URI Too Long](https://tools.ietf.org/html/rfc7231#section-6.5.12), not [413 Payload Too Large](https://tools.ietf.org/html/rfc7231#section-6.5.11) failures when we end up with too much information in the query string, correct?

The primary reason behind preferring `GET` over `POST` was for paging/following next links and/or location/operation-location headers. A client would not know which verb to use when following the link. 

A secondary reason is that GET is a safe and idempotent operations, which means that clients are more likely to retry on failed attempts.

A tertiary reason is the ability to cache responses (which `GET` generally allows for, but `POST` not so much). This may not be a concern for your specific API.

Will you ever provide both a query filter and a continuation token in the same request?

Reply:
I think we're in the common situation wherein the GET version is more "rest-standard" and the POST version will actually work better. If we're generating code, then users of the generated clients will have a better experience with the POST version, since they won't be exposed to the verb anyways. If we add both to the swagger, then both versions will show up in generated clients? Is it possible to document both, but only have one be the generated clients?

Reply:
+ @bterlson since he has expressed interest in similar discussions from a javascript client perspective. 

@andscho-msft, it should be possible to exclude a method from code generation, either through configuration or through an extension. I know that we have discussed exactly this scenario in the past, but I can't find the conclusion right now.

Reply:
Removed HTTP Get APIs for these operations until we have better solution for HTTP Get methods to handle 414 too long uri

Reply:
done